emitter/test/part2.test.ts

```diff
@@ -0,0 +1,212 @@
+    });
+
+    it("supports several event handlers", () => {
+        const trackedFunctionOne = vi.fn();
+        const trackedFunctionTwo = vi.fn();
+
+        emitter.on("Hello", trackedFunctionOne);
+        emitter.on("Hello", trackedFunctionTwo);
+
+        emitter.trigger("Hello");
+
+        expect(trackedFunctionOne).toHaveBeenCalledTimes(1);
+        expect(trackedFunctionTwo).toHaveBeenCalledTimes(1);
+    });
+
+    it("supports same function multiple times", () => {
+        const trackedFunction = vi.fn();
+
+        emitter.on("Hello", trackedFunction);
+        emitter.on("Hello", trackedFunction);
+
+        emitter.trigger("Hello");
+
+        expect(trackedFunction).toHaveBeenCalledTimes(2);
+    });
+
+    it("supports multiple event triggers same function", () => {
+        const trackedFunction = vi.fn();
+        const trackedFunctionNotToBeInvoked = vi.fn();
+
+        emitter.on("Hello", trackedFunction);
+        emitter.on("World", trackedFunction);
+
+        emitter.trigger("Hello");
+        emitter.trigger("World");
+
+        expect(trackedFunction).toHaveBeenCalledTimes(2);
+        expect(trackedFunctionNotToBeInvoked).toHaveBeenCalledTimes(0);
+    });
+
+    it("supports multiple event triggers different function", () => {
+        const trackedFunction = vi.fn();
+        const trackedFunction1 = vi.fn();
+
+        emitter.on("Hello", trackedFunction);
+        emitter.on("World", trackedFunction1);
+
+        emitter.trigger("Hello");
+        emitter.trigger("World");
+
+        expect(trackedFunction).toHaveBeenCalledTimes(1);
+        expect(trackedFunction1).toHaveBeenCalledTimes(1);
+    });
+
+    it("supports multiple events", () => {
+        const otherEventEmitter = new EventEmitter();
+        const trackedFunction = vi.fn();
+
+        emitter.on("Hello", trackedFunction);
+        otherEventEmitter.on("World", trackedFunction);
+
+        emitter.trigger("Hello");
+        otherEventEmitter.trigger("World");
+
+        expect(trackedFunction).toHaveBeenCalledTimes(2);
+    });
+
+    it("does not leak events between emitters", () => {
+        const otherEventEmitter = new EventEmitter();
+        const trackedFunction = vi.fn();
+
+        emitter.on("Hello", trackedFunction);
+        emitter.on("World", trackedFunction);
+        emitter.trigger("Hello");
+
+        otherEventEmitter.trigger("World");
+        expect(trackedFunction).toHaveBeenCalledTimes(1);
+    });
+});
+
+describe("Emitter event removal", () => {
+    let emitter;
+    let tracker;
+
+    beforeEach(() => {
+        emitter = new EventEmitter();
+        tracker = vi.fn();
+    });
+
+    it("can remove an event", () => {
+        const trackedFunction = vi.fn();
+
+        const off = emitter.on("inc", trackedFunction);
+        emitter.emit("inc");
+        off();
+        emitter.emit("inc");
+
+        expect(trackedFunction).toHaveBeenCalledTimes(1);
+    });
+
+    it("can remove and call off twice", () => {
```

Comment:
You already tested you can remove, this just checks calling twice won't cause any issues
```suggestion
    it("can call off twice", () => {
```